x/leverage/keeper/collateral.go

```diff
@@ -204,3 +205,75 @@ func (k *Keeper) checkCollateralShare(ctx sdk.Context, denom string) error {
 	}
 	return nil
 }
+
+// AvailableModuleCollateralLiquidity calculates the maximum available amount of uToken to withdraw
+// from the module given a token's denom. This amount is the sum of maximum available liquidity and
+// collateral from the module.
+func (k Keeper) AvailableModuleCollateralLiquidity(ctx sdk.Context, addr sdk.AccAddress, denom string) (
```

Comment:
```suggestion
func (k Keeper) moduleMaxWithdraw(ctx sdk.Context, addr sdk.AccAddress, denom string) (
```
Now that we know better the scope of our function, we can rename it. It might also be helpful to rename the existing `maxWithdraw` to `userMaxWithdraw` to increase clarity.

Reply:
**Optionally**, the `addr sdk.AccAddress, denom string` inputs could be replaced with `spendableUTokens sdk.Coin` (which contains the `uDenom` and the `userSpendableUtokens` amount we need later.)

In that case, the function signature might become more clear in how it doesn't depend on the address except for that one amount.